tests/utils.go

```diff
@@ -100,3 +110,38 @@ func GetKubeClientFromRESTConfig(config *rest.Config) (*kubernetes.Clientset, er
 
 	return coreClient, nil
 }
+
+// Creates a new namespace with a randomly generated name that starts with cdi-test-
+func GenerateNamespace(client *kubernetes.Clientset) *k8sv1.Namespace {
+	var namespace *k8sv1.Namespace
+	var err error
+	namespaceName := fmt.Sprintf("cdi-test-%s", strings.ToLower(util.RandAlphaNum(10)))
+	nsDef := &k8sv1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespaceName,
+		},
+	}
+	if wait.PollImmediate(2*time.Second, defaultTimeout, func() (bool, error) {
+		namespace, err = client.CoreV1().Namespaces().Create(nsDef)
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
+	}) != nil {
+		Fail("Unable to create namespace: " + namespaceName)
+	}
+	return namespace
+}
+
+// Destroys the passed in name space, be sure to clean any resources used before destroying the namespace.
+func DestroyNamespace(client *kubernetes.Clientset, namespace *k8sv1.Namespace) {
+	if wait.PollImmediate(2*time.Second, defaultTimeout, func() (bool, error) {
+		err := client.CoreV1().Namespaces().Delete(namespace.GetName(), &metav1.DeleteOptions{})
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
+	}) != nil {
+		Fail("Unable to remove namespace: " + namespace.GetName())
+	}
+}
```

Comment:
In k8s as part of the ns delete there is a loop to make sure it's deleted. Lots of nasty surprises if a ns gets reused. See [code](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#L1098-L1106)

Reply:
So that is a good question, it does take a little bit for the name space to get removed, but with the 10 character random part to the NS name getting the same one twice while its being deleted is highly unlikely. Now we can add the wait to eliminate it completely, but it would significantly slow down the tests, as after each test we would have to wait for the NS to be actually removed.

Reply:
@awels Right. For now it's fine as is. The deletion of the ns catches other namespaced resources that we either forget to delete or are slow getting deleted. I've seen many early k8s e2es fail timing out waiting for the ns to delete.

Reply:
We can account for the rare chance that the namespace already exists.

Just look at the error returned during the creation. If the error is "already exists" then we should generate another namespace to account for the collision. I really really doubt we'll ever hit that though.

Reply:
Added check for existing namespace, and it will generate another and try again.